lib/adf_diag.py

```diff
@@ -1080,3 +1108,227 @@ def derive_variables(self, vars_to_derive=None, ts_dir=None, overwrite=None):
+                    elif len(adf_file_list) > 1:
+                        if verbose > 0:
+                            print(
+                                f"WARNING: found multiple timeseries files {adf_file_list}. Continuing with best guess; suggest cleaning up multiple dates in ts dir"
+                            )
+                    else:
+                        if verbose > 0:
+                            print(
+                                f"WARNING: No files matching {case_name}.{hist_str}.{var} found in {adf_file_str}. Skipping"
+                            )
+                        continue  # skip this case/hist_str/var file
+                    adf_file = adf_file_list[0]
+
+                    # If freq is not set, it means we just started this hist_str. So check the first ADF file to find it
+                    hist_file_ds = xr.open_dataset(
+                        adf_file, decode_cf=False, decode_times=False
+                    )
+                    if "time_period_freq" in hist_file_ds.attrs:
+                        dataset_freq = hist_file_ds.attrs["time_period_freq"]
+                        if verbose > 2:
+                            print(f"time_period_freq attribute found: {dataset_freq}")
+                    else:
+                        if verbose > 0:
+                            print(
+                                f"ERROR: Necessary 'time_period_freq' attribute missing from {adf_file}"
+                            )
+                        continue
+
+                    found_strings = [
+                        word for word in freq_string_options if word in dataset_freq
+                    ]
+                    if len(found_strings) == 1:
+                        if verbose > 2:
+                            print(
+                                f"Found dataset_freq {dataset_freq} matches {found_strings}"
+                            )
+                    elif len(found_strings) > 1:
+                        if verbose > 0:
+                            print(
+                                f"WARNING: Found dataset_freq {dataset_freq} matches multiple string possibilities:{', '.join(found_strings)}"
+                            )
+                    else:
+                        if verbose > 0:
+                            print(
+                                "ERROR: None of the frequency options {freq_string_options} are present in the time_period_freq attribute {time_perido_freq}"
+                            )
+                            print("Skipping {adf_file}")
+                            freq = "frequency_missing"
+                        continue
+                    freq = found_strings[0]
+
+                    #
+                    # Destination file is MDTF directory and name structure
+                    #
+                    mdtf_dir = os.sep.join([mdtf_model_data_root, case_name, freq])
```

Comment:
I might change this line to the following:

```suggestion
                    mdtf_dir = os.path.join(mdtf_model_data_root, case_name, freq)
```